src/frontend/components/StudentLiveAdvertising/StudentLiveRegistration/index.tsx

```diff
@@ -41,39 +41,31 @@ export const StudentLiveRegistration = () => {
     return decodedJWT.user?.email ?? undefined;
   }, [decodedJWT]);
 
-  const [isLoading, setIsLoading] = useState(true);
-  const [liveSession, setLiveSession] = useState<Maybe<LiveSession>>();
+  const anonymousId = useMemo(() => {
+    let anonId: Maybe<string>;
+    if (!checkLtiToken(decodedJWT)) {
+      anonId = getAnonymousId();
+    }
 
-  useEffect(() => {
-    let canceled = false;
-    const checkRegistered = async () => {
-      let anonymousId: Maybe<string>;
-      if (!checkLtiToken(decodedJWT)) {
-        anonymousId = getAnonymousId();
-      }
-
-      let existingLiveSessions;
-      try {
-        existingLiveSessions = await getLiveSessions(anonymousId);
-      } catch (err) {
-        return;
-      }
-
-      if (canceled) {
-        return;
-      }
-      setIsLoading(false);
-      if (existingLiveSessions.count > 0) {
-        setLiveSession(existingLiveSessions.results[0]);
-      }
-    };
-
-    checkRegistered();
-    return () => {
-      canceled = true;
-    };
+    return anonId;
   }, [decodedJWT]);
 
+  const { liveSession, setLiveSession } = useLiveSession();
+  const { isLoading } = useLiveSessions(
+    { anonymous_id: anonymousId },
+    {
+      onSuccess: (data) => {
+        if (data.count > 0) {
+          setLiveSession(data.results[0]);
+        }
+      },
```

Comment:
Same question about the onError ? Previously the form was not displayed now it will be.